### PR TITLE
Add count to error message

### PR DESCRIPTION
**Pull request by nbigaouette**
Thanks for the crate! It's quite useful in a project of mine.

The `times()` method on a mock will assert that the expectation has been called a specific number of times.

Unfortunately, it does not print how many times it has been called! Only the expected count...

This makes it harder to diagnose what could have gone wrong in a test.

This PR adds the expectation call count to the panic message. For example:
```
MockFoo::baz: Expectation(<anything>) called 4 times which is more than expected 3
```
or
```
MockFoo::foo: Expectation(<anything>) called 0 time(s) which is fewer than expected 1
```

For this to work, I had to expose a new method `Times::count()`.

I left out any `Cargo.toml` version update as I am not sure of the project's policy on version. Let me know if a version bump is required.

### CODE REVIEW

File: mockall/src/lib.rs [hunk 1]
```diff
@@ -1472,7 +1472,11 @@ impl Times {
             if self.range.0.end == 1 {
                 Err("should not have been called".to_owned())
             } else {
-                Err(format!("called more than {} times", self.range.0.end - 1))
+                Err(format!(
+                    "called {} times which is more than expected {}",
```
Comment (asomers) [inline mockall/src/lib.rs:1476]: I think this would sound more natural:
```suggestion
                    "called {} times which is more than the expected {}",
```

File: mockall/tests/mock_struct.rs [hunk 2]
```diff
@@ -29,52 +29,45 @@ mod checkpoint {
     #[test]
     fn expect_again() {
         let mut mock = MockFoo::new();
-        mock.expect_foo()
-            .returning(|_| 5)
-            .times(1..3);
+        mock.expect_foo().returning(|_| 5).times(1..3);
```
Comment (asomers) [inline mockall/tests/mock_struct.rs:32]: Could you please revert the unrelated formatting changes to this file?

**Review by asomers — CHANGES_REQUESTED** [review]
Hey, thanks for the contribution!  It generally looks good.  I just had a few inline comments.  As for the version number, don't worry about it.  I bump it whenever I make a release.  Also, changes to mockall normally require an entry in the CHANGELOG.  But this PR is an exception.  Changes to a human readable error message don't need an entry there.

File: mockall/tests/mock_struct.rs (hunk 2 above)
Comment (nbigaouette) [reply mockall/tests/mock_struct.rs]: I've re-done commit 65bfd7d as 9af1d7d which should not include format change. Sorry it slipped in...

File: mockall/src/lib.rs (hunk 1 above)
Comment (nbigaouette) [reply mockall/src/lib.rs]: I agree! Fixed in d1fffe3.
